plugins/outputs/cloudwatchlogs/internal/pusher/target.go

```diff
@@ -109,26 +133,89 @@ func (m *targetManager) createLogStream(t Target) error {
+	for target := range m.retentionPolicyUpdates {
+		err := m.processWithRetry(target)
 		if err != nil {
-			// since this gets called both before we start pushing logs, and after we first attempt
-			// to push a log to a non-existent log group, we don't want to dirty the log with an error
-			// if the error is that the log group doesn't exist (yet).
-			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
-				m.logger.Debugf("Log group %v not created yet: %v", t.Group, err)
-			} else {
-				m.logger.Errorf("Unable to put retention policy for log group %v: %v ", t.Group, err)
+			m.logger.Errorf("Failed to update retention policy for target %v: %v", target, err)
+		}
+	}
+}
+
+func (m *targetManager) processWithRetry(target Target) error {
+	var err error
+	for attempt := 0; attempt < maxAttempts; attempt++ {
+		err = m.checkAndUpdate(target)
+		if err == nil {
+			return nil
+		}
+
+		m.logger.Debugf("retrying to update retention policy for target (%v) %v: %v", attempt, target, err)
+		delay := m.calculateBackoff(attempt)
+		time.Sleep(delay)
+	}
+	return fmt.Errorf("failed to update retention policy after %d attempts: %w", maxAttempts, err)
+}
+
+func (m *targetManager) checkAndUpdate(target Target) error {
+	currentRetention, err := m.describeLogGroupRetention(target)
+	if err != nil {
+		return err
+	}
+
+	if currentRetention != target.Retention {
+		m.logger.Debugf("updating retention policy for target %v: %v", target, err)
+		return m.updateRetentionPolicy(target)
+	}
+
+	return nil
+}
+
+func (m *targetManager) describeLogGroupRetention(target Target) (int, error) {
+	input := &cloudwatchlogs.DescribeLogGroupsInput{
+		LogGroupNamePrefix: aws.String(target.Group),
+	}
+
+	output, err := m.service.DescribeLogGroups(input)
+	if err != nil {
+		return 0, fmt.Errorf("describe log groups failed: %w", err)
+	}
+
+	for _, group := range output.LogGroups {
+		if *group.LogGroupName == target.Group {
+			if group.RetentionInDays == nil {
+				return 0, nil
 			}
-		} else {
-			m.logger.Debugf("successfully updated log retention policy for log group %v", t.Group)
+			return int(*group.RetentionInDays), nil
 		}
 	}
+
+	return 0, fmt.Errorf("log group %v not found", target.Group)
+}
+
+func (m *targetManager) updateRetentionPolicy(target Target) error {
+	input := &cloudwatchlogs.PutRetentionPolicyInput{
+		LogGroupName:    aws.String(target.Group),
+		RetentionInDays: aws.Int64(int64(target.Retention)),
+	}
+
+	_, err := m.service.PutRetentionPolicy(input)
+	if err != nil {
+		return fmt.Errorf("put retention policy failed: %w", err)
+	}
+
+	return nil
+}
+
+func (m *targetManager) calculateBackoff(attempt int) time.Duration {
+	delay := baseDelay
+	for i := 0; i < attempt && delay < maxDelay; i++ {
+		delay *= 2
+	}
+	if delay > maxDelay {
+		delay = maxDelay
+	}
+
+	jitter := time.Duration(float64(delay) * (float64(time.Now().UnixNano()%100) / 500.0 * jitterFactor))
```

Comment:
nit: Would expect jitter to have a degree of randomness. https://aws.amazon.com/builders-library/timeouts-retries-and-backoff-with-jitter/